api/v1alpha1/qperf_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// QperfPort is the TCP port where the qperf server and client listens
+const QperfPort = 19765
+
+// QperfConfigurationSpec contains configuration parameters
+// with scheduling options for the both the qperf client
+// and server instances.
+type QperfConfigurationSpec struct {
+	PodConfigurationSpec `json:",inline"`
+
+	// HostNetwork requested for the qperf pod, if enabled the
+	// hosts network namespace is used. Default to false.
+	// +optional
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+}
+
+// QperfSpec defines the Qperf Benchmark Stone which
+// consist of server deployment with service definition
+// and client pod.
+type QperfSpec struct {
```

Comment:
QPerf supports udp mode. Would it make sense to add that here?

Reply:
As of my understanding qperf communicates everything on the `listen_port` and the tests are running on a different port. It can be a problem if any communication limited between pods on the Kubernetes cluster. I have to test it anyway.

Reply:
Tested with UDP, worked.

CR:
```yaml
kind: Qperf
metadata:
  name: qperf-sample
spec:
  image:
    name: xridge/qperf:0.4.11-r0
  options: "--verbose --time 10"
  tests:
  - udp_bw
  - udp_lat
```

Output:
```
udp_bw:
    send_bw         =   543 MB/sec
    recv_bw         =   227 MB/sec
    msg_rate        =  6.93 K/sec
    time            =    10 sec
    send_cost       =  1.98 sec/GB
    recv_cost       =  2.99 sec/GB
    send_cpus_used  =   107 % cpus
    recv_cpus_used  =  67.8 % cpus
udp_lat:
    latency        =  52.7 us
    msg_rate       =    19 K/sec
    time           =    10 sec
    loc_cpus_used  =  17.7 % cpus
    rem_cpus_used  =  18.6 % cpus
```